tests/Nox.Types.Tests/Types/EncryptedText/EncryptedTextTests.cs

```diff
@@ -1,11 +1,180 @@
 ﻿// ReSharper disable once CheckNamespace
+
+using System.Security.Cryptography;
+using FluentAssertions;
+
 namespace Nox.Types.Tests.Types;
 
 public class EncryptedTextTests
 {
+    private readonly EncryptedTextTypeOptions _typeOptions;
+    private const string PlainText = "plain text";
+    private readonly Aes _aesAlg;
+
+    public EncryptedTextTests()
+    {
+        _aesAlg = Aes.Create();
+
+        _typeOptions = new EncryptedTextTypeOptions
+        {
+            PublicKey = _aesAlg.Key,
+            EncryptionAlgorithm = EncryptionAlgorithm.Aes,
+            Iv = _aesAlg.IV
+        };
+    }
+
+    [Fact]
+    public void Constructor_ReturnsEncryptedValue()
+    {
+        var encryptedText = EncryptedText.FromPlainText(PlainText, _typeOptions);
+
+        encryptedText.Should().NotBeNull();
+        encryptedText.Value.Should().NotBeNull().And.NotBeEmpty();
+    }
+
+    [Fact]
+    public void Decryption_ReturnsOriginalValue()
+    {
+        var encryptedText = EncryptedText.FromPlainText(PlainText, _typeOptions);
+        var decryptedText = encryptedText.DecryptText(_typeOptions);
+
+        PlainText.Should().Be(decryptedText);
+    }
+
+    [Fact]
+    public void SameEncryptParameters_Returns_SameEncryptedValue()
+    {
+        var encryptedText = EncryptedText.FromPlainText(PlainText, _typeOptions);
+        var encryptedText2 = EncryptedText.FromPlainText(PlainText, _typeOptions);
+
+        encryptedText.Should().NotBeNull();
+        encryptedText2.Should().NotBeNull();
+        encryptedText.Value.Should().Equal(encryptedText2.Value);
+    }
+
+    [Fact]
+    public void DifferentEncryptParameters_Returns_DifferentEncryptedValue()
+    {
+        // Change key
+        _aesAlg.GenerateKey();
+        var encryptOptions1 = new EncryptedTextTypeOptions
+        {
+            EncryptionAlgorithm = EncryptionAlgorithm.Aes,
+            PublicKey = _aesAlg.Key,
+            Iv = _aesAlg.IV
+        };
+
+        // Change IV
+        _aesAlg.GenerateIV();
+        var encryptOptions2 = new EncryptedTextTypeOptions
+        {
+            EncryptionAlgorithm = EncryptionAlgorithm.Aes,
+            PublicKey = _aesAlg.Key,
+            Iv = _aesAlg.IV
+        };
+
+        var encryptedText = EncryptedText.FromPlainText(PlainText, _typeOptions);
+        // Different key
+        var encryptedText1 = EncryptedText.FromPlainText(PlainText, encryptOptions1);
+        // Different IV
+        var encryptedText2 = EncryptedText.FromPlainText(PlainText, encryptOptions2);
+
+        encryptedText.Value.Should().NotEqual(encryptedText1.Value);
+        encryptedText1.Value.Should().NotEqual(encryptedText2.Value);
```

Comment:
This can be added also as an assertion:
`encryptedText.Value.Should().NotEqual(encryptedText2.Value);`